src/pages/Content/index.js

```diff
@@ -84,247 +42,311 @@ async function getUserInfo() {
+        //Render the tooltip
+        endX = e.pageX;
+        endY = e.pageY;
+        console.log('start x is ', startY);
+        console.log('end x is ', endY);
+
+        const realStartX = Math.min(startX, endX);
+        const realendX = Math.max(startX, endX);
+
+        const realStartY = Math.min(startY, endY);
+        const realEndY = Math.max(startY, endY);
+        lastSelection = selection;
+        lastSelectionObj = window.getSelection();
+
+        renderToolTip(
+          (realendX - realStartX) / 2 + realStartX,
+          realStartY - (realEndY - realStartY) / 2,
+          selection
+        );
+      } else {
+        closeToolTip();
+      }
+    });
+
+    //Highlight options
+    document.addEventListener('click', (e) => {
+      const parentIdName = e.target.parentNode.getAttribute('id');
+      const currentID = e.target.getAttribute('id');
+      const range = lastSelectionObj != null ? lastSelectionObj.getRangeAt(0) : null;
+      console.log("Range object:", range)
+      const payload = {
+        url: currentURL,
+        userID: currentUserInfo.id,
+        highlightSelection: rangySerializer.serializeRange(range) // Serializes the range into a string to store in DB
+      }
+
+      if (parentIdName == 'highlight' || currentID == 'highlight') {
+
+        /*
+        ISSUE 3: 
+        May need to update the method of update to the DB because someone may just KEEP pushing highlights
+        We need a check before making an API call
+        
+        Might need to check this out: https://github.com/Phildesro123/Reliant#secrets
+        */
+        axios.post('http://localhost:4000/api/websites/addHighlights', payload).then((res) => {
+          console.log(res);
+          highlightText('#ffc107', range);
+        })
+        closeToolTip();
+      } else if (parentIdName == 'smile' || currentID == 'smile') {
+
+        highlightText('#28a745', range);
+        closeToolTip();
+      } else if (parentIdName == 'frown' || currentID == 'frown') {
+        highlightText('#dc3545', range);
+        closeToolTip();
+      } else if (parentIdName == 'comment' || currentID == 'commment') {
+        highlightText('#dc3545', range, true);
+        //Youssef's comment
+        closeToolTip();
+      } else if (parentIdName == 'note' || currentID == 'note') {
+        highlightText('blue', range, true);
+        // Implement note
+        closeToolTip();
+      }
+    });
+
+
+    const highlightText = (color, range, underline=false) => {
+      var mark = document.createElement('mark')
+      if (underline) {
+        console.log("UNDERLINING")
+        mark = document.createElement('u');
+        mark.style.textDecoration = "underline";
+        mark.style.textDecorationColor = color;
+        mark.style.textDecorationThickness = ".2rem";
+        mark.style.textDecorationSkipInk = "none"
+      } else {
+        mark.style.backgroundColor = color;
+        mark.style.textDecoration = 'none';
+      }
+
+      /*
```

Comment:
### ISSUE 4   @avin1100 
Need a check if the common ancestor of the range is a mark.

Ideally, we should "overwrite" the existing highlight with the newer one

- Need checks for nested highlights
- As of now, if you "update" (highlight a highlight) a highlight - the mark doesn't get update but instead a new mark tag gets nested in it
        

Right now, im suspecting the extractContents as the culprit because it's extract the contents of the range
        If the range has a mark it'll just add the mark to the newly created mark
        For more reference of the Range API checkout: https://developer.mozilla.org/en-US/docs/Web/API/Range